tools/driver/p4c_src/main.py

```diff
@@ -0,0 +1,225 @@
+                        help="Treat subsequent input files as having type language.",
+                        action="store", default="p4-16")
+    parser.add_argument("-I", dest="search_path",
+                        help="Add directory to include search path",
+                        action="append", default=[])
+    parser.add_argument("-o", dest="output_directory",
+                        help="Write output to the provided path",
+                        action="store", metavar="PATH", default=".")
+    parser.add_argument("--target-help", dest="show_target_help",
+                        help="Display target specific command line options.",
+                        action="store_true", default=False)
+
+    parser.add_argument("source_file", nargs='?', help="Files to compile", default=None)
+
+    # many more options
+    opts = parser.parse_args()
+    source = opts.source_file
+
+    if opts.show_version:
+        print("p4c %s" % (p4c_src.__version__))
+        sys.exit(0)
+
+    # target-arch-vendor, e.g.
+    # bmv2-*-p4org
+    # bmv2-ssa-p4org
+    # ebpf-psa-p4org
+    triplet = opts.backend.split('-')
+    if (len(triplet) != 3):
+        print "Invalid target-arch-vendor triplet."
+        sys.exit(1)
+
+    if not source:
+        parser.error('No input specified.')
+
+    # load supported configuration
+    cfg_files = glob.glob("{}/*.cfg".format(os.environ['P4C_CFG_PATH']))
+    cfg = config.Config(config_prefix = "p4c")
+    for cf in cfg_files:
+        if opts.debug:
+            print 'loading config {}'.format(cf)
+        cfg.load_from_config(cf, opts.output_directory, opts.source_file)
+
+    if opts.show_target_help:
+        print "Supported backends in \"target-arch-vendor\" triplet:"
+        for target in cfg.target:
+            print target
+        sys.exit(0)
+
+    backend = None
+    for triplet, _ in cfg.steps.iteritems():
+        regex = triplet.replace('*', '[a-zA-Z0-9*]*')
+        pattern = re.compile(regex)
+        if (pattern.match(opts.backend)):
+            backend = triplet
+            break
+    if backend == None:
+        print "Unknown backend:", opts.backend
+        sys.exit(1)
+
+    commands['preprocessor'] = []
+    commands['compiler'] = []
+    commands['assembler'] = []
+    commands['linker'] = []
+
+    for option in opts.preprocessor_options:
+        commands["preprocessor"] += shlex.split(option)
+
+    for option in opts.compiler_options:
+        commands["compiler"] += shlex.split(option)
+
+    for option in opts.assembler_options:
+        commands["assembler"] += shlex.split(option)
+
+    for option in opts.linker_options:
+        commands["linker"] += shlex.split(option)
+
+    # set output directory
+    if not os.path.exists(opts.output_directory):
+        os.makedirs(opts.output_directory)
+
+    commands['preprocessor'].insert(0, cfg.get_preprocessor(backend))
+    commands['compiler'].insert(0, cfg.get_compiler(backend))
```

Comment:
if the compiler is just built in the current directory and one does not have the current dir in the path, then it can not find the compiler, unless run with:
```
PATH=.:$PATH ./p4c -v ../testdata/p4_16_samples/arith-bmv2.p4
```
I wonder whether we should force the current directory in the path.